questionary/styles.py

```diff
@@ -0,0 +1,12 @@
+from typing import List
+from typing import Optional
+
+import prompt_toolkit.styles
+
+from questionary.constants import DEFAULT_STYLE
+
+
+def merge_styles_default(styles: List[Optional[prompt_toolkit.styles.Style]]):
+    filtered_styles: list[prompt_toolkit.styles.BaseStyle] = [DEFAULT_STYLE]
+    filtered_styles.extend([s for s in styles if s is not None])
+    return prompt_toolkit.styles.merge_styles(filtered_styles)
```

Comment:
What's the reasoning for this? Just to remove any `None` styles?

Reply:
I believe prompt_toolkit has some improper type hinting. `merge_styles` can certainly handle `None` elements, but it's not in their type hints. So this fixes that so mypy passes.

Reply:
Can you pop a comment in the source code explaining the reasoning?